samples/net/echo: Prune OpenThread tests

andyross opened the pull request:
(Workaround for #12206 pending west integration of external dependencies)

Currently the OpenThread integration wants to download (from github)
configure and build the whole library at the app level, and we're
doing it four times every time we run sanitycheck.  And it's a slow
process that doesn't parallelize well.  It's adding almost a whole
minute to my ~6 minute build times on a very parallel host.

Eventually this external dependency handling will be integrated in
west, but for now let's run just one OpenThread test.

Note that this thread also removes the needless "test_kw41z" tests,
which look like mistakes to me.  That platform is already whitelisted
in the main test (named "test"), and this version includes no variant
configuration, it's just a duplication.

Signed-off-by: Andy Ross <andrew.j.ross@intel.com>

nashif reviewed (changes requested):
we can also mark those tests as slow and just run them daily basis.
```
  slow: true

```

Commit:
samples/net/echo: Remove duplicate test case

These tests were defining a "test_kw41z" case, but that platform is
already whitelisted in the main ("test") case, and this one defines no
different behavior.  It's just a dupe.

Signed-off-by: Andy Ross <andrew.j.ross@intel.com>

Commit:
samples/net/echo: Mark OpenThread test cases as "slow"

The current OpenThread integration wants to download (from github) and
configure the OpenThread source tree every time the app is build.  We
don't want to be doing this in quadruplicate every time sanitycheck is
run.

Signed-off-by: Andy Ross <andrew.j.ross@intel.com>

Commit:
scripts/sanitycheck: Don't build "slow" tests either

Extend the "slow" flag on tests to inhibit building too.  The original
assumption was that building would be fast but running slow, but now
We have tests using a component (OpenThread) that wants to pull and
build software from github for every app.

Signed-off-by: Andy Ross <andrew.j.ross@intel.com>

andyross commented:
I didn't even know "slow" existed.  Yeah, that's much better.

But, heh, sanitycheck didn't actually do it that way.  It would build the slow tests always, it would only inhibit running them.  So there's a final patch that modifies sanitycheck to skip building those tests too (I figure if @nashif suggested it, this won't break anything).

andyross commented:
recheck
